### PR TITLE
Added controller init

### PR DESCRIPTION
**Pull request by jellekralt**
With some projects, I want to load some logic when the server starts. Because I don't want to modify the express.js file too heavily with project specific stuff, I've added globbing of the controllers and a check if that module has an 'init' function. If so, the init function will be executed when the server starts. In this way, you can load stuff when the server starts while the logic stays in the controllers.

### CODE REVIEW

**Comment by roieki** [comment]
https://en.wikipedia.org/wiki/Bullet-point_engineering

**Comment by sielay** [comment]
To init routing you load all controller modules already. Why you can't put you init code there? Meaning while you declare your controller. Just an idea? This way you will load only linked controller and (more important) you won't create all requires twice.

**Comment by ilanbiala** [comment]
I would actually put all the init stuff in a file a return a promise so that you can do stuff with the info before the server starts. So if I have to check if any users exist, I would do anything necessary in the promise and handle that promise in either `server.js` or `express.js`.

File: config/express.js [hunk 1]
```diff
@@ -33,6 +33,14 @@ module.exports = function(db) {
 		require(path.resolve(modelPath));
 	});
 
+	// Globbing controllers files
+	config.getGlobbedFiles('./app/controllers/*.js').forEach(function(controllerPath) {
+		var controller = require(path.resolve(controllerPath));
+		if('init' in controller) {
```
Comment (ilanbiala) [inline config/express.js]: Why pass in `app`? Can you show a case where it is necessary? Why not also just do `if (controller.init)`?

**Comment by lirantal** [comment]
I'm closing as this didn't get any further action on.
@jellekralt you're welcome to re-open with a possible 0.4.0 PR if you think it's still relevant.

**Pull request closed without merging**